viewsb/packet.py

```diff
@@ -733,9 +758,133 @@ def from_setup_data(cls, setup_data, **fields):
     def summarize(self):
         return "control request setup transaction for {} request".format(self.request_direction.name)
 
+    @staticmethod
+    def _decode_descriptor_type(descType):
```

Comment:
Any reason this couldn't be a dictionary?

```python
descriptor_type = {
    0: 'invalid',
    1: 'device',
    2: 'configuration',
    ...
}

descriptor_type.get(desc_type, f'unknown ({descType:#x})')
```

The same applies to the other methods, but this is the most obvious case.

Also, nitpick :stuck_out_tongue:. Use snake case (`desc_type`) instead of camel case (`descType`) for variables.

Reply:
Also, numbers >= 32 are context dependent, so you can't just decode them to string like that without knowing where they are in the descriptors.

Reply:
thanks for the point out that my own style had bled into this code.. I'll go fix that..

I thought about using a dictionary, but both because of the context sensitivity and to try and keep this part fast.. I decided not to.. at least, that I recall.

I can change it to use a dictionary. @smunaut: Thought some of those were fixed in translation like the HID ones, but if I'm wrong I'm happy to change this to be sensitive to some further information.

Reply:
Nope...

33 is also DFU functional descriptor for instance if the interface class is DFU.

In some classes like UAC / UVC you have 33-37 are also "Class Specific Device/Conf/String/Intf/EP" descriptors where an added bDescriptorsubtype gives more detail about what follows.

Reply:
Don't get me wrong, the code is perfectly fine as-is :sweat_smile:. This is a code style question, hence subjective, and ultimately up to the maintainers. I was just raising the question in case you hadn't really thought about that :blush:

Reply:
Thanks for the correction @smunaut - I'll look at revising this based on the presence of the proper class indicators in the descriptor (if we can) or otherwise falling it back from the core fixed-meaning ones to simple numeric decoders as I'm not sure we have the descriptor information available at this point in the decoding process

Reply:
I've now corrected the naming convention violations and class-specific translations.
If Kate/Qyriad wish these translations to be done by dictionary, then I will switch them over.

Reply:
Yeah I think I'd prefer these be dictionaries.